### PR TITLE
#51 Add debug logging settings

### PR DESCRIPTION
**Pull request by AnnaYatsun1**
Add new Class  LoggingAspect with  3 aspects (@Before, @Around,   @After) and add in pom dependency

### CODE REVIEW

**Comment by AnnaYatsun1** [comment]
change class withn't loombok

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package io.github.externschool.planner.aop;
+
+
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.*;
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: Please don't use wildcard imports

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.aspectj.lang.reflect.MethodSignature;
+
+
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: one blank line is OK, the second one is unnecessary

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+public class LoggingAspect {
+    private Logger log = LoggerFactory.getLogger(getClass());
+
+    @Pointcut("within(io.github.externschool.planner.controller..*) || within(io.github.externschool.planner.service..*) ")
+    private void allMethods() {
+
+    }
+
+        @Before("allMethods()")
+    public void log(JoinPoint joinPoint) {
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: Annotation starts at the same column as the method declaration, please remove extra spaces before '@Before'

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        for (int i = 0; i < arguments.length; i++) {
+            System.out.println(arguments[i]);
+        }
+    }
+    @Around("allMethods()")
+    public Object logBusinessMethods(ProceedingJoinPoint call) throws Throwable {
+        if (!log.isDebugEnabled()) {
+            return call.proceed();
+        } else {
+
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: what is this blank line for?

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            Object[] args = call.getArgs();
+            String message = call.toShortString();
+            log.info("{} called with args '{}'!", message, Arrays.deepToString(args));
+            Object result = null;
+
+
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: The second blank line is not necessary here. Single line could be used as a logic blocks delimiter or if you want to emphasize something like a return statement for example. Extra lines make your code hard to read.

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+
+  @After("allMethods()")
+   public void logAfter(JoinPoint joinPoint) {
+            log.info("after: "+joinPoint.getTarget().getClass().getSimpleName()+" "+joinPoint.getSignature().getName());
+
+    }
+
+
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: We put one blank line after closing curly braces, not between

File: src/main/java/io/github/externschool/planner/aop/LoggingAspect.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+
+
+
```
Comment (Benkoff) [inline src/main/java/io/github/externschool/planner/aop/LoggingAspect.java]: Read the recommended style guide, please (shared link in Slack)

**Pull request merged**